---
bool new_state
string message